addons/popochiu/engine/objects/room/popochiu_room.gd

```diff
@@ -287,6 +287,26 @@ func set_active_walkable_area(walkable_area_name: String) -> void:
 
 
 # ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ PRIVATE ░░░░
+
+func _update_character_scale(chr):
+	if chr.on_scaling_region:
+		var polygon_range = chr.on_scaling_region['polygon_bottom_y'] - chr.on_scaling_region['polygon_top_y']
+		var scale_range = chr.on_scaling_region['scale_bottom'] - chr.on_scaling_region['scale_top']
+
+		var position_from_the_top_of_region = chr.position.y-chr.on_scaling_region['polygon_top_y']
+
+		var scale_for_position = (
```

Comment:
- Same as previous comment. I would prefer to avoid having lines of code that go beyond line 100.
- When possible, define the type of the variables you're using.

```suggestion
		var polygon_range: float = (
			chr.on_scaling_region['polygon_bottom_y'] - chr.on_scaling_region['polygon_top_y']
		)
		var scale_range: float = (
			chr.on_scaling_region['scale_bottom'] - chr.on_scaling_region['scale_top']
		)

		var position_from_the_top_of_region: float = (
			chr.position.y-chr.on_scaling_region['polygon_top_y']
		)

		var scale_for_position: float = (
```